test/NuGet.Core.Tests/NuGet.Protocol.Tests/LocalResourceTests/LocalPackageSearchResourceTests.cs

```diff
@@ -586,18 +587,57 @@ public async Task LocalPackageSearchResource_FileSourceAsync()
         }
 
         [Fact]
-        public async Task LocalPackageSearch_SearchAsync_WithCancellationToken_ThrowsAsync()
+        public async Task LocalPackageSearch_SearchAsync_WithCancellationToken_ImmediatelyThrowsAsync()
         {
             using (var root = TestDirectory.Create())
             {
+                // Arrange
                 var localResource = new FindLocalPackagesResourceV2(root);
                 LocalPackageSearchResource resource = new LocalPackageSearchResource(localResource);
 
-                await Assert.ThrowsAsync<OperationCanceledException>(
+                // Act & Assert
+                await Assert.ThrowsAsync<TaskCanceledException>(
                     async () => await resource.SearchAsync("", null, 0, 1, NullLogger.Instance, new CancellationToken(canceled: true)));
             }
         }
 
+        [Fact]
+        public async Task LocalPackageSearch_SearchAsync_SlowLocalRepository_WithCancellationToken_ThrowsAsync()
```

Comment:
I meant to leave a comment here.

I don't think this test is helpful right now. 

The cancellation token is raised by test code, not the product code, so you're effective testing the test code.

Reply:
It's best I can do for now, idea is same. I'll try to come up with something better for next PR.

Reply:
Let me rephrase. 
I don't think it adds anything, and I think it should be removed.

You can add an equivalent test in the future PR you mentioned.

Reply:
Sure, removed.